# Chemistry I
## Course code: CH 1101
- Credits: 3
- Program: BTech
- Course Status: Core

## Tutorial Sheet 2

## Working with `scipy.constants`

In [2]:
from scipy.constants import c             # speed of light
from scipy.constants import m_e           # mass of electron
from scipy.constants import m_p           # mass of proton
from scipy.constants import h             # Planck's constant
from scipy.constants import N_A           # Avogadro constant
from scipy.constants import k             # Boltzmann constant
from scipy.constants import e             # elementary charge
from scipy.constants import Rydberg as Ry # Rydberg constant
from scipy.constants import eV            # one electron volt in Joules

print( 'c = {} m s^-1'.format( c ) )
print( 'm_e = {} kg'.format( m_e ) )
print( 'm_p = {} kg'.format( m_p ) )
print( 'h = {} J s'.format( h ) )
print ( 'N_A = {} mol^-1'.format (N_A) )
print( 'k = {} J K^-1'.format( k ) )
print( 'e = {} C'.format(e) )
print( 'Ry = {} m^-1'.format(Ry) )
print( 'eV = {} J'.format(eV) )

c = 299792458.0 m s^-1
m_e = 9.1093837015e-31 kg
m_p = 1.67262192369e-27 kg
h = 6.62607015e-34 J s
N_A = 6.02214076e+23 mol^-1
k = 1.380649e-23 J K^-1
e = 1.602176634e-19 C
Ry = 10973731.56816 m^-1
eV = 1.602176634e-19 J


#### Physical constants¶
`scipy.constants` also contains a `physical_constants` database (Python dictionary)
that gives a larger number of constants along with their their units.

In [3]:
from scipy.constants import physical_constants

print( physical_constants['speed of light in vacuum'] )
print( physical_constants['electron mass'] )
print( physical_constants['proton mass'] )
print( physical_constants['Planck constant'] )
print( physical_constants['Avogadro constant'] )
print( physical_constants['Boltzmann constant'] )
print( physical_constants['elementary charge'] )
print( physical_constants['proton mass'] )
print( physical_constants['Rydberg constant'] )
print( physical_constants['electron volt'] )

(299792458.0, 'm s^-1', 0.0)
(9.1093837015e-31, 'kg', 2.8e-40)
(1.67262192369e-27, 'kg', 5.1e-37)
(6.62607015e-34, 'J Hz^-1', 0.0)
(6.02214076e+23, 'mol^-1', 0.0)
(1.380649e-23, 'J K^-1', 0.0)
(1.602176634e-19, 'C', 0.0)
(1.67262192369e-27, 'kg', 5.1e-37)
(10973731.56816, 'm^-1', 2.1e-05)
(1.602176634e-19, 'J', 0.0)


Each entry in the dictionary returns three components: (1) value, (2) unit (as a string), and (3) experimental uncertainty.

In [4]:
import math
from math import pi

In [5]:
print(pi)

3.141592653589793


### Question 1
What kind of molecular processes are studied using each of the following kind of electromagnetic radiations?
- (a) X-rays
- (b) UV-Visible light
- (c) Infrared radiation
- (d) Microwaves
- (e) Radio waves

### Answer:
- (a) X-rays: Core electronic transitions

- (b) UV-Visible light: Valence electronic transitions

- (c) Infrared radiation: Molecular vibrations

- (d) Microwaves: Molecular rotations

- (e) Radio waves: Nuclear spin states detection


### Question 2
When light of wavelength 400 nm passes through 3.5 mm of a solution of an absorbing substance at a concentration 0.67 mmol/lit, the transmission is 65.5 percent. Calculate the molar absorption coefficient of the solute at this wavelength.
### Solution:
$$A = -log~T = \epsilon~c~l \quad {\rm and} \quad T=\frac{I}{I_0}$$
- A: Absorbance
- T: Transmittance
- &epsilon;: molar absorption coefficient
- l: path length which is the cell thickness
- c: concentration of solution
- I<sub>0</sub>: incident light intensity
- I: transmitted light intensity

$$\epsilon = \frac{-log~T}{c~l}$$ Let's first calculate T = 65.5/100 = 0.655

$$\Rightarrow \epsilon = \frac{-log~T}{c~l} = \frac{-log~0.655}{(3.5~mmol~lit^{-1})\times(3.5~mm)} = 0.0784~ mmol^{-1}~lit~mm^{-1} $$

In [6]:
conc  = 0.67                                           # c, mmol/lit
trans = 65.5/100                                       # T
cell  = 3.5                                            # l, mm

epsilon_400 = - math.log10(trans) / ( conc * cell)     # lit mol^-1 mm^-1
print( "Molar absorption coefficient = {:.2e} mmol^-1 lit mm^-1".format( epsilon_400 ) )

Molar absorption coefficient = 7.84e-02 mmol^-1 lit mm^-1


### Question 3
The molar absorption coefficient of a solute at 440 nm is 323 lit mol<sup>-1</sup> cm<sup>-1</sup>. When light of that wavelength passes through a 7.50 mm cell containing a solution of the solute, 52.3 per cent of the light was absorbed. What is the concentration of the solution?
### Solution:
$$ c  = \frac{-log~T}{\epsilon~l}$$ Let's first calculate T = (100-52.3)/100 = 0.477

$$\Rightarrow c = \frac{-log~T}{\epsilon~l} = \frac{-log~0.477}{(323~lit~mol^{-1}cm^{-1})\times(0.75~cm)} = 0.00133~mol~lit^{-1} $$

In [7]:
epsilon_440 = 323                   # eplison, lit mol^-1 cm^-1
cell = 0.75                         # l, cm

trans = (100 - 52.3)/100            # T

conc = - math.log10(trans) / (epsilon_440 * cell)
print( "Concentration of the solution = {:.2e} mol/lit".format( conc ) )

Concentration of the solution = 1.33e-03 mol/lit


### Question 4
An electronic ground state and excited state in a molecule are separated by an energy gap of 10000 cm<sup>-1</sup>. Estimate the fraction of molecules that will be found in the excited state at room temperature (27°C). What would be the fraction of molecules in excited state at the same temperature if the energy separation is 100 MHz (typical energy separation between nuclear spin states).

### Solution:
$$\frac{N_f}{N_i} = e^{-\Delta E/k_B T}$$

##### Learning concept:
A certain fraction of molecule can be found in the excited state if the energy gap is comparable to k<sub>B</sub>T

 $$ k_B T= (1.38\times 10^{-23}~J~K^{-1})\times(300~K) = 4.14\times 10^{-24}~J$$
 
$$\Delta E = h\nu \quad {\rm and} \quad \nu = \frac{c}{\lambda} \Rightarrow \Delta E = \frac{hc}{\lambda} $$

$$ \Delta E  = (6.626\times 10^{-34}~J~Hz^{-1}) \times (3.0\times 10^{8}~m~s^{-1})$$

$$= (6.626\times 10^{-34}~J~s^{-1}) \times (3.0\times 10^{8}~m~s^{-1})$$

$$= (6.626\times 10^{-34}~J) \times (3.0\times 10^{8}~m)$$

$$ \Rightarrow \frac{1}{m} = (6.626\times 10^{-34}~J) \times (3.0\times 10^{8}) $$

$$ \Rightarrow \frac{1}{(100~cm)} = (6.626\times 10^{-34}~J) \times (3.0\times 10^{8}) $$

$$ \Rightarrow 0.01~cm^{-1} = (6.626\times 10^{-34}~J) \times (3.0\times 10^{8}) $$

$$ \Rightarrow cm^{-1} = \frac{(6.626\times 10^{-34}~J) \times (3.0\times 10^{8})}{0.01} $$

### Lets try to solve it for 10000 cm<sup>-1</sup>

$$ \Rightarrow 10000~cm^{-1} = 10000\times\frac{(6.626\times 10^{-34}~J) \times (3.0\times 10^{8})}{0.01} = 1.99\times 10^{-19} J $$

Now, 
$$\frac{N_f}{N_i} = e^{-\frac{1.99\times 10^{-19} J}{4.14\times 10^{-24}~J}} \sim 0$$

#### This means that most of the molecules will be in the ground state.

In [11]:
T = 300                                 # Temperature, T = 27+273 = 300 K
kbt = k*T
                                        # RT = kB * T * NA  [JK^(-1) * K]
                                                                               
kbt_j = kbt; 

print( "kB*T = {:.2e} J".format( kbt_j ) )

kB*T = 4.14e-21 J


In [12]:
#e = hc/lambda
cmi = 10000
cmi2j = h * c / 0.01         # conversion factor from cm^-1 to Joule                

de_j = cmi * cmi2j


print( "Energy gap = {:.2e} J".format( de_j ) )
nf = math.exp(-de_j/kbt_j)
print( "Molecules in excited state = {:.2e}".format( nf ) )

Energy gap = 1.99e-19 J
Molecules in excited state = 1.48e-21


### Lets try to solve it for 100 MHz

$$\Delta E = h\nu = h~[J~Hz^{-1}] \times \nu~[Hz^{-1}] =  h\nu ~ [J~Hz^{-1} \times Hz^{-1}] = h\nu~[J] $$

$$ 100~MHz = 100\times 10^{6} Hz $$

$$ (6.626\times 10^{-34} \times 100\times 10^{6})~[J] = 6.63\times 10^{-26} J $$

Now, 
$$\frac{N_f}{N_i} = e^{-\frac{6.63\times 10^{-26} J}{4.14\times 10^{-24}~J}} \sim 1.0 $$

#### This simply means that most of the molecules will be excited.

In [13]:
#e = hc/lambda
mhz = 100
de_j = h * mhz * 10**6 # conversion factor from Hz to Joule                


print( "Energy gap = {:.2e} J".format( de_j ) )
nf = math.exp(-de_j/kbt_j)
print( "Molecules in excited state = {:.2e}".format( nf ) )

Energy gap = 6.63e-26 J
Molecules in excited state = 1.00e+00
